docs_src/design/ucr/0002-Device-Parent-Child-Relationships.md

```diff
@@ -0,0 +1,115 @@
+
+It is also necessary to indicate which device is the "main" or "publisher" device (ie, the gateway device), 
+as any devices without a configured relationship can be inferred to be children of that device.
+
+It is frequently a pattern in data servers to "walk the device tree", starting with the main device, then 
+recursively processing its direct child devices, and then the child devices (if any) of those devices, until
+all devices have been processed. This is normally part of the initialization of device data for a server,
+since the parent must be processed and initialized before its child devices. Consequently, there is a need
+for a means to answer the question "What are the child devices (if any) of device x.y.z?"; this is commonly done
+either with the device structure listing its children, or by providing a query that can answer this question.
+
+### Extensions to the main Use Case
+1. Some services add "devices" which have no physical counterpart, eg, for an NTP client service where the "device" 
+serves simply as a container for the Resources necessary to configure and report the status of the service.
+In these cases, it would be helpful for the other services if it described itself as something like a "system" device,
+meaning one that doesn't have a physical (south-bound or hardware) counterpart.
+2. Multi-level systems, such as a "gateway of gateways", which aggregate a large number of devices. The parent
+property will be useful for assisting to arrange the devices in a hierarchy that corresponds to their physical hierarchy. 
+In this situation, the parent property of a lower-level gateway will indicate the upper gateway it is connected to.
+
+### Existing solutions
+The Device structure in Eaton's legacy products indicated this parent-child relationship bidirectionally: each device indicated its parent device (if any) with one field, and its child devices (if any) with a list of IDs.
+
+The Device structure in Eaton's cloud solution is a "DeviceTree", which is a recursive, hierarchial structure of the connected devices, starting with the "publisher" device and its first-level child devices.
+
+There is the BACnet "virtual routed devices" model, but I would not recommend it, as it is too artificial for this  simple relationship.
+
+The existing EdgeX UIs group devices by their Device Service, which is a good approach for simple devices without children of their own, but fails if those devices have child devices too.
+
+### Requirements
+1. Each device should have a property to indicate which other device is its parent device.
+2. There should be one "main" or "publisher" device, and it should indicate this via a property instead of
+indicating a parent device (since it has no parent).
+3. If a device does not indicate which is its parent device, then it shall be inferred that its parent is the main device. (This helps with backwards compatability.)
```

Comment:
2. and 3. might be able to be combined to `Some means to indicate whether this device is a "main" device, such as a special property or not defining the parent device.`
Also, I thought the name "root" device might be easier understandable than "main" or "publisher" device.

Reply:
> 2. and 3. might be able to be combined to `Some means to indicate whether this device is a "main" device, such as a special property or not defining the parent device.`
>    Also, I thought the name "root" device might be easier understandable than "main" or "publisher" device.

I think 2 is about having a main device, while 3 is about defaulting the parents to the main device.